components/studio/bin/hab-studio.sh

```diff
@@ -1160,6 +1097,30 @@ set_libexec_path() {
   return 0
 }
 
+# If `file_path` is not present in the studio, copy in a minimal
+# default version from the studio package's `defaults` directory.
+copy_minimal_default_file_if_not_present() {
+    file_path="${1}"
+    if [ ! -f "${HAB_STUDIO_ROOT}${file_path}" ]; then
+        copy_minimal_default_file "${file_path}"
+    fi
+}
+
+copy_minimal_default_file() {
+    file_path="${1}"
+    defaults_path="$($bb dirname "${libexec_path}")/defaults"
+    if [ -n "$VERBOSE" ]; then
+        echo "> Creating minimal ${file_path}"
+    fi
+    if [ -f "${defaults_path}${file_path}" ]; then
+        $bb cp "${defaults_path}${file_path}" "${HAB_STUDIO_ROOT}${file_path}"
+    else
+        echo
+        echo "Tried to copy default file for '${file_path}', but could not find one! Please report this error."
```

Comment:
Not quite sure... if this happens, it means that we've either left a file out of the package, or have otherwise messed up our code. However, this is being checked at runtime, and not at buildtime, and thus any error could make it impossible to enter a studio.

Leaving it as a notification could allow users to still add their own file. Of course, as I type this, we could still exit the studio, and users could "patch" things by dropping a file into the unpacked package directory under `/hab/pkgs` to fix things temporarily.

Neither seems ideal. I'm open to suggestions on the best way to handle this. Ideally, we'd have some kind of check at build time that could detect if we were trying to operate on a "default" file that wasn't actually present.

Reply:
I'm more in favor of fail-fast. Yes, a notification would allow users to still add their own file, but I think it's highly unlikely that notification will be seen and acted upon. And if not, who knows how things might misbehave later?

Plus, as you say, it would still be possible to fix things up by modifying the unpacked package directory, so I don't think making this fatal puts us in a worse position.

Reply:
@baumanj Good point. Check out https://github.com/habitat-sh/habitat/pull/6319/commits/8543d6a83319486339013ca5bc7f1f9666f4851c